iothub/device/src/Edge/DeviceClientFactory.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using Microsoft.Azure.Devices.Client.HsmAuthentication;
+
+namespace Microsoft.Azure.Devices.Client.Edge
```

Comment:
The code looks good. My concern with this would be with the API - now we have 2 "Factory" patterns - one on the DeviceClient itself (it has Create, CreateFromConnectionString), and then we also have this DeviceClientFactory that has a create method for a different purpose. 
The 2 should be unified. We should move the API methods in this file to DeviceClient which can then call this internal type to construct the DeviceClient.

Reply:
Because this factory is specific to Edge, I think would be better to keep it separate from DeviceClient Create methods. When we have a separate moduleSDK, we can move it there.

Reply:
I agree with Varun.  Please move it to the DeviceCLient

---
In reply to: [184792231](https://github.com/Azure/azure-iot-sdk-csharp/pull/462#discussion_r184792231) [](ancestors = 184792231)